src/uu/touch/src/touch.rs

```diff
@@ -83,32 +83,35 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     ) {
         (Some(reference), Some(date)) => {
             let (atime, mtime) = stat(Path::new(reference), !matches.get_flag(options::NO_DEREF))?;
-            if let Some(offset) = parse_relative_time(date) {
-                let mut seconds = offset.whole_seconds();
-                let mut nanos = offset.subsec_nanoseconds();
-                if nanos < 0 {
-                    nanos += 1_000_000_000;
-                    seconds -= 1;
+            match humantime_to_duration::from_str(date) {
+                Ok(offset) => {
+                    let mut seconds = offset.whole_seconds();
+                    let mut nanos = offset.subsec_nanoseconds();
+                    if nanos < 0 {
+                        nanos += 1_000_000_000;
+                        seconds -= 1;
+                    }
+
+                    let ref_atime_secs = atime.unix_seconds();
+                    let ref_atime_nanos = atime.nanoseconds();
+                    let atime = FileTime::from_unix_time(
+                        ref_atime_secs + seconds,
+                        ref_atime_nanos + nanos as u32,
+                    );
+
+                    let ref_mtime_secs = mtime.unix_seconds();
+                    let ref_mtime_nanos = mtime.nanoseconds();
+                    let mtime = FileTime::from_unix_time(
+                        ref_mtime_secs + seconds,
+                        ref_mtime_nanos + nanos as u32,
+                    );
+
+                    (atime, mtime)
+                }
+                Err(..) => {
```

Comment:
Happy to accept this as a first version, but the date parsing would need to be part of the humantime crate and ignoring the error here is not quite right. We could open an issue for this once this PR is merged.